javascript/0041-first-missing-positive.js

```diff
@@ -0,0 +1,47 @@
+ /**
+   * Time O(n) | Space O(1)
+ * @param {number[]} nums
+ * @return {number}
+ */
+var firstMissingPositive = function(nums) {
+    //-1 1 3 4
+    const swap = (a,b) => {
+        const temp = nums[a];
+        nums[a] = nums[b];
+        nums[b] = temp;
+    }
+    // swap positions
+    for(let i = 0; i < nums.length; i++) {
+        const el = nums[i];
+        const chair = el - 1;
+        if(el >= 1 &&  el <= nums.length + 1 && nums[chair] !== el) {
```

Comment:
Suggestion: Decalre variable describing the condition

```js
const is...
if (is...)

const has...
if (has...)

const should..
if (should...)
```

Reply:
I don't get it. What do you mean. Can you kindly update the code and let me know what do you mean? I actually followed this solution https://www.youtube.com/watch?v=aBhla9jSAeg&t=13s&ab_channel=CodewithAlisha I have used variable as chair because it made sense to me.

Reply:
https://github.com/neetcode-gh/leetcode/pull/2560#discussion_r1248973743